tests/variable_definitions/test_vardef_write_template.py

```diff
@@ -0,0 +1,151 @@
+    base_path.mkdir(parents=True, exist_ok=True)
+
+    with patch.object(Path, "cwd", return_value=base_path), pytest.raises(
+        VardefFileError,
+        match="VardefFileError: File not found at file path: unknown file path",
+    ):
+        Vardef.write_template_to_file()
+
+
+@pytest.mark.usefixtures("set_temp_workspace")
+def test_write_template_random_dir_work_dir(tmp_path: Path):
+    base_path = tmp_path / "statistics/a/work"
+    base_path.mkdir(parents=True, exist_ok=True)
+    with patch.object(Path, "cwd", return_value=base_path):
+        file_path = Vardef.write_template_to_file()
+        assert file_path.exists()
+
+
+@pytest.mark.usefixtures("set_temp_workspace")
+def test_write_template_from_current():
+    with patch.object(Path, "cwd", return_value=Path("./")):
+        file_path = Vardef.write_template_to_file()
+        assert file_path.exists()
+
+
+def test_write_template_file_exists(mocker):
+    mocker.patch(
+        "dapla_metadata.variable_definitions.utils.variable_definition_files._model_to_yaml_with_comments",
+        side_effect=FileExistsError("File already exists"),
+    )
+
+    with pytest.raises(
+        VardefFileError,
+        match="File already exists and can not be saved: unknown file path",
+    ):
+        Vardef.write_template_to_file()
+
+
+def test_write_template_permission_error(mocker):
+    """Test that _model_to_yaml_with_comments raises VardefFileError when a PermissionError occurs."""
+    mocker.patch(
+        "dapla_metadata.variable_definitions.utils.variable_definition_files._model_to_yaml_with_comments",
+        side_effect=PermissionError("Permission denied"),
+    )
+
+    with pytest.raises(VardefFileError, match="Permission denied for file path"):
+        Vardef.write_template_to_file()
+
+
+def test_write_template_time_stamp_error(mocker):
+    mocker.patch(
+        "dapla_metadata.variable_definitions.utils.variable_definition_files._get_current_time",
+        side_effect=UnknownTimeZoneError("Unknown timezone"),
+    )
+
+    with pytest.raises(VardefFileError, match="Timezone is unknown"):
+        Vardef.write_template_to_file()
+
+
+def test_yaml_serialization_error(mocker):
+    """Test that write_template_to_file raises VardefFileError when yaml serialization fails."""
+    mocker.patch(
+        "dapla_metadata.variable_definitions.utils.variable_definition_files._model_to_yaml_with_comments",
+        side_effect=YAMLError("Failed to serialize YAML"),
+    )
+
+    with pytest.raises(VardefFileError, match="Not possible to serialize yaml"):
+        Vardef.write_template_to_file()
+
+
+def test_write_template_to_yaml_eof_error(mocker):
+    mocker.patch(
+        "dapla_metadata.variable_definitions.utils.variable_definition_files._model_to_yaml_with_comments",
+        side_effect=EOFError("Unexpected end of file"),
+    )
+
+    with pytest.raises(VardefFileError, match="Unexpected end of file"):
+        Vardef.write_template_to_file()
```

Comment:
Could these be parameterized since they're all different cases of the same test? I don't think it's necessary/desirable to test the text in the message, I think it should be possible to change exception text without breaking tests.